crates/admin/src/cluster_controller/service.rs

```diff
@@ -305,6 +307,14 @@ impl<T: TransportConnect> Service<T> {
                         warn!("Could not trim the logs. This can lead to increased disk usage: {err}");
                     }
                 }
+                Ok(_) = tail_udpates_watcher.changed() => {
+                    // we clone here to avoid holding the read lock in the watcher
+                    // more than necessary.
+                    // todo(azmy): do we really need to keep the updates inside an Arc?
+                    // the on_logs_tail_updates should return quick enough.
```

Comment:
I guess the argument is to be more defensive and keep the `Arc` - even if we documented the requirement to keep the borrow lifetime short, it would be easy to miss making changes to faraway code in the future. Seems worth the cost/benefit tradeoff to me to add a comment why we did it this way, and keep as-is?